bot/seasons/season.py

```diff
@@ -1,38 +1,45 @@
 import asyncio
 import datetime
 import importlib
+import inspect
 import logging
 import pkgutil
 from pathlib import Path
+from typing import List, Optional, Type, Union
 
+import async_timeout
+import discord
 from discord.ext import commands
 
-from bot.constants import Client, Roles
+from bot.constants import Channels, Client, Roles, bot
 from bot.decorators import with_role
 
 log = logging.getLogger(__name__)
 
 
-def get_seasons():
+def get_seasons() -> List[str]:
     """
-    Returns all the Season objects
-    located in bot/seasons/
+    Returns all the Season objects located in bot/seasons/
     """
+
     seasons = []
 
-    for module in pkgutil.iter_modules([Path('bot', 'seasons')]):
+    for module in pkgutil.iter_modules([Path("bot", "seasons")]):
         if module.ispkg:
-            seasons.append(module[1])
-
+            seasons.append(module.name)
     return seasons
```

Comment:
Couldn't we have this `yield` the modules?

Reply:
For what benefit.

Reply:
Also this feels unrelated to the current PR.